### ISSUE TITLE
dartdevk Function objects comparing issue

### ISSUE THREAD

**Issue by sgrekhov** [dart-lang/sdk#42895]
The following test fails on dartdevk and works as expected on other platforms
```dart
import "../../../Utils/expect.dart";

class C {
  m() {
  }
}

String foo(x1, y1, {x2 : "x2", y2}) {
  return "$x1$y1$x2$y2";
}

String bar(x1, y1, {x2 : "x2", y2}) {
  return "$x1$y1$x2$y2";
}

main() {
  Function f1 = foo;
  Function f2 = foo;
  Function b1 = bar;
  Function b2 = bar;

  Expect.isTrue(f1 == f2);
  Expect.isTrue(b1 == b2);
  Expect.isFalse(f1 == b1);

  Function m1 = new C().m;
  Function m2 = new C().m;
  Expect.isFalse(m1 == m2);
}
```
dartdevk output is
```
Runtime error:
TypeError: f1._equals is not a function
```

**Comment by sigmundch** [comment]
It appears this issue relates to how we [emit the name of the equality operator][1]

Note that if the type `f1` is changed to `Object` instead of `Function`, then it works as expected.

For context, DDC may generate names for methods using a simple member name like `._equals` or a symbolized reference like `[$_equals]`. Because symbols are private and not visible outside our code, we use them to extend native objects (like primitive types, DOM objects, arrays, etc) with Dart methods and operators.

When generating calls to these members we also choose between the two names. The symbolized name always works, but it can be slower, so we try to choose the simple name if we can. We only choose the simple name when we know that the member exists in the target class, but we use the symbolized name when we are not sure or when we know that are calling a native object.

We correctly treat the `Object` type as an interface type that could be implemented by native types, but we don't seem to be recording that properly for the `Function` type.

[1]: https://github.com/dart-lang/sdk/blob/master/pkg/dev_compiler/lib/src/kernel/compiler.dart#L4773

**Comment by nshahan** [comment]
Fixed by ffd84c47b8bf3529328f6d4f2801ec9b31b00b9a.
